torch/nested/_internal/triton_kernels/grouped_mm.py

```diff
@@ -0,0 +1,237 @@
+    assert not tensor_b.is_nested
+    group_size = a_offsets.size(0)
+
+    assert tensor_b.is_contiguous()
+
+
+    assert a_values.dim() == 2
+
+    B, K, N = tensor_b.shape
+
+    c_values = a_values.new_empty((a_values.size(0), N))
+    config_key = gen_config_key(a_values, a_offsets, max_M, tensor_b)
+    if config_key in BEST_CONFIGS:
+        best_config = BEST_CONFIGS[config_key]
+    else:
+        current_timestamp = datetime.datetime.now()
+        print_prefix = current_timestamp.strftime("%Y-%m-%d %H:%M:%S")
+        print(print_prefix, f"Don't have a config for config_key {config_key}. Need to run autotuning.")
+        best_ms, best_config = None, None
+        all_configs = gen_configs()
+        # Use a random order to increase chance of finding a good config early.
+        random.shuffle(all_configs)
+        # with concurrent.futures.ProcessPoolExecutor(max_workers=1) as executor:
+        for i, config in enumerate(all_configs):
+            current_timestamp = datetime.datetime.now()
+            print_prefix = current_timestamp.strftime("%Y-%m-%d %H:%M:%S")
+            print(print_prefix, end=' ')
+            sys.stdout.flush()
+            ms = do_bench(group_gemm_fn_kernel, [a_values, a_offsets, max_M, tensor_b, c_values, config], best_ms)
+            # if best_ms is None:
+            #     ms = do_bench(group_gemm_fn_kernel, [a_values, a_offsets, max_M, tensor_b, c_values, config], best_ms)
+            # else:
+            #     assert best_ms is not None
+            #     ms_future = executor.submit(do_bench, group_gemm_fn_kernel, [a_values, a_offsets, max_M, tensor_b, c_values, config], best_ms)
+            #     try:
+            #         timeout = ((100 * best_ms) / 1000)
+            #         print(print_prefix, "timeout: ", timeout)
+            #         ms = ms_future.result(timeout=timeout)
+            #     except concurrent.futures.TimeoutError:
+            #         print(print_prefix, f"Timeout with config number {i} out of {len(all_configs)}: {config}")
+            #         ms_future.cancel()
+            #         continue
+            print(f"     ms: {ms} with config number {i} out of {len(all_configs)}: {config}")
+            if not isinstance(ms, str) and best_ms is None:
+                best_ms, best_config = ms, config
+                print(print_prefix, f"best_ms: {best_ms} with config: {best_config}")
+            if not isinstance(ms, str) and best_ms > ms:
+                best_ms, best_config = ms, config
+                print(print_prefix, f"best_ms: {best_ms} with config: {best_config}")
+        if best_config is None:
+            raise ValueError("Could not find valid config.")
+        BEST_CONFIGS[config_key] = best_config
+        print(f"Found config {best_config} with ms {best_ms}.")
+
+    return group_gemm_fn_kernel(a_values, a_offsets, max_M, tensor_b, c_values, best_config)
+
+@torch.library.impl(lib, "group_gemm_fn", "Meta")
```

Comment:
Guard on multipy

Reply:
@zou3519 - Is this guard also needed with the custom op API?

Reply:
Driss is referring to a bug we had. I fixed it (https://github.com/pytorch/pytorch/issues/137536). So there should be no need to guard on multipy.